File: medium/2683. Neighboring Bitwise XOR/main.py
from typing import List


class Solution:
    def doesValidArrayExist(self, derived: List[int]) -> bool:
        original = [0] * (len(derived) + 1)

        original[0] = 0
        for i in range(0, len(derived)):
            original[i+1] = derived[i] ^ original[i]

        check_for_zero = (original[0] == original[len(original) - 1])

        original[0] = 1
        for i in range(0, len(derived)):
            original[i+1] = derived[i] ^ original[i]

        check_for_one = (original[0] == original[len(original) - 1])

        return (check_for_zero or check_for_one)

File: medium/2683. Neighboring Bitwise XOR/main.ts
const doesValidArrayExist = (derived: number[]): boolean => {
  let original = new Array<number>(derived.length + 1);

  original[0] = 0;
  for (let i = 0; i < derived.length; i++) {
    original[i + 1] = derived[i] ^ original[i];
  }

  let checkForZero = (original[0] == original[original.length - 1]);

  original[0] = 1;
  for (let i = 0; i < derived.length; i++) {
    original[i + 1] = derived[i] ^ original[i];
  }

  let checkForOne = (original[0] == original[original.length - 1]);

  return checkForZero || checkForOne;
};
